tests/onnx/test_onnx_v2.py

```diff
@@ -262,6 +264,8 @@ def _onnx_export(self, test_name, name, model_name, feature, onnx_config_class_c
 
         model_class = FeaturesManager.get_model_class_for_feature(feature)
         config = AutoConfig.from_pretrained(model_name)
+        if model_name == "microsoft/deberta-v2-xlarge":
+            config.type_vocab_size = 2
```

Comment:
@JingyaHuang A weird thing about the DeBERTa model is that it is the only model that by default takes token_type_ids in its forward method (the tokenizer returns this in the dictionary by default) which the model config by default sets the vocab size to 0 (this basically ignores that value in the forward method).

This has a few downstream side-effects, the ONNX graph perceives this as meaning the model by default only takes 2 inputs which causes the test to fail because we attempt to pass all 3 items in the ONNX input. What I have done is in the ONNX config I override the value to be 2 so that the graph is constructed taking 3 inputs (the default for the tokenizer but not the model) and had to hack in this statement to initialize the model config to expect token_type_ids as well.

This feels pretty bad. I think that the tokenizer should by default only return 2 items unless the model config has type_vocab_size > 0. However, this would be a larger change to the DeBERTa classes and I was unsure if that was desired.

Reply:
Hi @sam-h-bean whether the exported ONNX model needs 2 or 3 inputs depends on whether you've included `token_type_ids` in the custom OnnxConfig, as the dummy inputs used for tracing will [look into the custom OnnxConfig](https://github.com/huggingface/transformers/blob/v4.19.4/src/transformers/onnx/convert.py#L139) to decide what generated inputs will be considered for export. 

As the tokenizer of DeBERTa will return `token_type_ids` by default, we suggest including `token_type_ids` into its OnnxConfig even if it is optional(it will be [ignored](https://github.com/huggingface/transformers/blob/main/src/transformers/models/deberta_v2/modeling_deberta_v2.py#L823) if `type_vocab_size` is set to 0)

Reply:
Yeah I can do that but here we still need to set the `AutoConfig` value some way because the token type embedding is initialized on creation here. So if I just have it overwritten in the onnx config it fails for there being no token embedding layer since by default there is none. So how should I change the config here dynamically?